File: 643_MaximumAverageSubarrayI.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
  var sum = nums.slice(0, k).reduce((num1, num2) => +num1 + +num2);
  var tmp = sum;

  for (var i = 1; i <= nums.length - k; i += 1) {
    tmp += (nums[i + k - 1] - nums[i - 1]);
    sum = Math.max(sum, tmp);
  }

  return sum / k;
};

File: 643.MaximumAverageSubarrayI.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int result = 0;
        int tmp = 0;
        for (int i = 0; i < k; i += 1) {
            result += nums[i];
        }
        tmp = result;
        for (int i = 0; i < nums.length - k; i += 1) {
            tmp -= nums[i];
            tmp += nums[i + k];
            result = Math.max(tmp, result);
        }
        return result / 1.0 / k;
    }
}